### ISSUE TITLE
Display report in logs detailing installation status of requested operators

### ISSUE THREAD

**Issue by bitscuit** [IBM/operand-deployment-lifecycle-manager#623]
/kind feature

**Describe the solution you'd like**
After creating an `OperandReqest`, the status of each requested operator is only shown in the `OperandRequest` status field - ODLM logs do not indicate whether installation of these operators was successful or not. This leads to extra debugging time when installation is not successful because there could be multiple `OperandRequests`, so would need to look through all of them to determine which operator did not install properly.

ODLM logs should show a table with each row containing an operator from the `OperandRequest` and an indication successful or unsuccessful installation, e.g.:

Operator | Status
--- | ---
ibm-mongodb-operator | Installed
ibm-iam-operator | Error

**Anything else you would like to add:**
An improved solution would be to also include installation status of CRs from each operator, e.g.:

Service | Status
--- | ---
ibm-mongodb-operator | Installed
ibm-mongodb-operator/MongoDB/ibm-mongodb | Installed
ibm-iam-operator | Installed
ibm-iam-operator/Authentication/example-authentication | Installed
ibm-iam-operator/OIDCClientWatcher/example-oidcclientwatcher | Error

But implementing this is challenging since not all CRs offer good status information

**Comment by gyliu513** [comment]
It would be great if we can also add `reason` here as well.

Service | Status|Reason
--- | ---|---
ibm-mongodb-operator | Installed |
ibm-mongodb-operator/MongoDB/ibm-mongodb | Installed |
ibm-iam-operator | Installed |
ibm-iam-operator/Authentication/example-authentication | Installed |
ibm-iam-operator/OIDCClientWatcher/example-oidcclientwatcher | Error |

**Comment by gyliu513** [comment]
Related with https://github.com/IBM/operand-deployment-lifecycle-manager/issues/622

**Comment by Daniel-Fan** [comment]
I am wondering that whether it is possible to show the installation status in the `events` section of the ODLM instead of logs of pod.
Currently we did not have info listed in the `events` section.

**Comment by gyliu513** [comment]
I was also wondering if we can print out the log periodically, so that we can always get the real time progress, and after install finished, print out a summary of install.

**Comment by bitscuit** [comment]
@Daniel-Fan 
> 
> 
> I am wondering that whether it is possible to show the installation status in the `events` section of the ODLM instead of logs of pod.
> Currently we did not have info listed in the `events` section.

I think events section should be reserved for information about ODLM pod itself, not information about the work it is doing

@gyliu513 for time elapsed, do you mean how long it took for operand to be installed/errored?

> 
> 
> I was also wondering if we can print out the log periodically, so that we can always get the real time progress, and after install finished, print out a summary of install.

I think @horis233 mentioned we can print the log whenever ODLM tries to reconcile an OperandRequest.

**Comment by gyliu513** [comment]
> @gyliu513 for time elapsed, do you mean how long it took for operand to be installed/errored?

@bitscuit Yes . But if we can support print the log periodically, then the state may not `installed/errored`, if maybe `installing/pending` etc, so for this case, the `time elapsed` just means the time for this state.

**Comment by gyliu513** [comment]
@horis233 @bitscuit how is this going?

**Comment by horis233** [comment]
@bitscuit Do you have any progress or a plan for this issue?

**Comment by bitscuit** [comment]
Sorry, I have not looked at this recently because of various release work. I can try to include it in the next dev sprint, but if anything more urgent comes up, this will be the first thing to be put on hold.

**Comment by horis233** [comment]
@bitscuit no problem.

@gyliu513 We can try to commit it in 2Q.
